change label to p to prevent label click-through onto input

135ze opened the pull request:
# Notion Ticket

[Clicking the box should open dropdowns/activate the inputs, clicking the text above the dropdowns/input should not](https://www.notion.so/uwblueprintexecs/Clicking-the-box-should-open-dropdowns-activate-the-inputs-clicking-the-text-above-the-dropdowns-in-23e10f3fb1dc80c8a3edf77a94674079)

## Summary & Review Focus

<!-- Briefly describe the implementation, key changes, and areas needing review -->

The "click fowarding" bug occurs because it is built-in to HTML labels, probably for accessibility: https://stackoverflow.com/a/49615087

I am able to change this behaviour by switching the label to be interpreted `as="p"`. To keep accessibility, I added aria-labelledby, which allows screen readers to still pick up on the purpose of the input without the unwanted behaviour.

To verify, run locally and try to click the labels and validate that the dropdowns do not get activated. 

Inputs still work because the fields are still populated correctly:

<img width="1388" height="846" alt="Screenshot 2025-08-19 222206" src="https://github.com/user-attachments/assets/8cfc13fd-590d-4e3f-a6be-502176144145" />

## Checklist

- [x] PR title is descriptive and in imperative tense
- [x] Commit messages are descriptive, atomic, and follow best practices
- [x] Linter(s) have been run
- [x] Requested reviews from the PL and relevant team members

135ze committed:
change label to p to prevent label click-through onto input

ChinemeremChigbo commented on src/components/absences/modals/edit/EditAbsenceForm.tsx:
This line and (and also lines 363 and 389) still renders<FormLabel htmlFor=...> for Room Number, Reason of Absence, and Notes, so clicking those headings continues to focus/activate the fields in the edit modal.

ChinemeremChigbo commented on src/components/absences/modals/DateOfAbsence.tsx:
```diff
@@ -65,7 +65,7 @@ export const DateOfAbsence: React.FC<DateOfAbsenceProps> = ({
   return (
     <FormControl isRequired isInvalid={!!error}>
-      <FormLabel sx={{ display: 'flex' }}>
+      <FormLabel as="p" id="dateOfAbsenceLabel" sx={{ display: 'flex' }}>
```
This hardcodes the id to dateOfAbsenceLabel; when both declare and edit modals mount together this produces duplicate IDs, so the aria-labelledby on the second date field points to the first label and breaks its own accessible name. Consider using React's useId() inside DateOfAbsence or passing an id down via props.

135ze replied:
Updated to add useId in this and the other forms too

135ze committed:
add useId to label ids, fix labels in EditAbsenceForm

ChinemeremChigbo committed:
Improve accessibility for teacher selection fields

Added unique aria-labelledby attributes and label IDs to the AdminTeacherFields and SearchDropdown components to enhance accessibility. Also fixed minor formatting in DeclareAbsenceForm. These changes ensure better screen reader support for form fields.
